### PR TITLE
Fixed cannot save in TreeView mode

### PR DESCRIPTION
**Pull request by VicenteMovai**
[FP-2754](https://movai.atlassian.net/browse/FP-2754): Control + S doesn't save flow
  - There was a random popup when switching between TreeView and Default mode
  - Forcing the focus (keybind) when we switch modes
  - Fix app blowing up when we saved in TreeView (not related to this issue, just needed to do File -> Save Document)
  
[FP-2653](https://movai.atlassian.net/browse/FP-2653): Nodes disappear after updating a flow parameter in tree view mode
  - When we save a Flow the onTemplateUpdate is triggered, and the onFlowUpdate is called, this function isn't ready to deal when the flow is in treeview mode and was deleting the Nodes in the TreeView.

[FP-2754]: https://movai.atlassian.net/browse/FP-2754?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FP-2653]: https://movai.atlassian.net/browse/FP-2653?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by duartecoelhomovai — APPROVED** [review]

File: src/plugins/DocManager/DocManager.js [hunk 1]
```diff
@@ -283,14 +283,17 @@ class DocManager extends IDEPlugin {
...
-
+      
       this.call(PLUGINS.ALERT.NAME, PLUGINS.ALERT.CALL.SHOW, {
         message: i18n.t(ERROR_MESSAGES.FAILED_TO_SAVE),
         severity: ALERT_SEVERITIES.ERROR
       });
     }
 
-    callback?.(returnMessage);
+    // callback is false when we try to save in TreeView mode
```
Comment (diasdm) [inline src/plugins/DocManager/DocManager.js:293]: Can we always send undefined so that this condition is not needed?

**Review by diasdm — CHANGES_REQUESTED** [review]

File: src/plugins/DocManager/DocManager.js (hunk 1 above)
Comment (VicenteMovai) [reply src/plugins/DocManager/DocManager.js:293]: We tried to find were the `callback = false` was coming from, but didn't succeed. But maybe @quirinpa or @EdwardAngeles can help find

**Comment by diasdm** [comment]
Found where false was being sent as callback :+1:

**Review by diasdm — CHANGES_REQUESTED** [review]
There's an exception in the updateNode changes:
```
ReferenceError: _node is not defined
    at GraphTreeView._this.updateNode (index.js:22481:1)
    at index.js:21658:1
    at Array.forEach (<anonymous>)
    at GraphBase.updateNodes (index.js:21657:1)
    at GraphBase.onFlowUpdate (index.js:21588:1)
    at GraphBase.onTemplateUpdate (index.js:21610:1)
    at index.js:39437:1
    at Map.forEach (<anonymous>)
    at DocManager.onDocumentUpdate (index.js:39436:1)
    at DBSubscriber.overwriteDoc (index.js:44944:1)
```

**Comment by VicenteMovai** [comment]
Fixed the exception @diasdm , but it wasn't trivial because it was linked to this issue [FP-2653](https://movai.atlassian.net/browse/FP-2653). I fixed it for now, but the saves aren't perfect in TreeView mode. Because imagine if you change a node template it updates correctly on the default view, but it doesn't on the Treeview. This was working on 2.3.1 but not in 2.4. But this is another regression bug

**Comment by diasdm** [comment]
> Fixed the exception @diasdm , but it wasn't trivial because it was linked to this issue [FP-2653](https://movai.atlassian.net/browse/FP-2653). I fixed it for now, but the saves aren't perfect in TreeView mode. Because imagine if you change a node template it updates correctly on the default view, but it doesn't on the Treeview. This was working on 2.3.1 but not in 2.4. But this is another regression bug

Thank you for the effort :bow:

**Review by diasdm — APPROVED** [review]

**Pull request merged**